opennms-services/src/main/java/org/opennms/netmgt/snmpinterfacepoller/SnmpPoller.java

```diff
@@ -240,10 +242,22 @@ protected void scheduleNewSnmpInterface(String ipaddr) {
      * <p>scheduleExistingSnmpInterface</p>
      */
     protected void scheduleExistingSnmpInterface() {
-        
-    	for (OnmsIpInterface iface : getNetwork().getContext().getPollableNodes()) {
-            schedulePollableInterface(iface);    		
-    	}
+        ExecutorService executor = getExecutorService();
+        for (OnmsIpInterface iface : getNetwork().getContext().getPollableNodes()) {
+            executor.execute(() -> schedulePollableInterface(iface));
+        }
+        executor.shutdown();
```

Comment:
I'm not doing this in an IDE so I'm not certain it's right, but I'd be doing something like...

```java
        ExecutorService executor = getExecutorService();
        List<Future<?>> futures = getNetwork().getContext().getPollableNodes()
            .stream()
            .map(iface -> executor.submit(() -> schedulePollableInterface(iface)).collect(Collectors.toList());
        CompletableFuture.allOf(futures).join();
```

Reply:
...which really could just be changed so `schedulePollableInterface(iface)` uses the executor and returns the future, instead, now that I think of it, since there's no reason not to do it in a thread pool for _all_ scheduled stuff, not just new existing ones on init... lol

Reply:
Good idea I will look into youre example and try it out!